src/StreamSender.mo

```diff
@@ -144,12 +138,11 @@ module {
     /// any new chunks.
     ///
     /// `#ready` means that the stream sender is ready to send a chunk.
-
     public func status() : Status {
       if (shutdown) return #shutdown;
       if (stopped) return #stopped;
       if (paused) return #paused;
-      if (concurrentChunks == settings_.windowSize) return #busy;
+      if (isBusy()) return #busy;
```

Comment:
I preferred the low-level condition because all the `isXXX` function are an optional add-on interface. We even thought of changing that interface at some point, hence better to not depend on it.